### PR TITLE
BuildXL now builds cleanly on the mac, and demos work as well

### PR DESCRIPTION
**Pull request by blufugen**

### CODE REVIEW

File: Public/src/Sandbox/MacOs/BuildXLSandbox/Src/BuildXLSandbox.cpp [hunk 1]
```diff
@@ -87,7 +87,7 @@ void BuildXLSandbox::InitializePolicyStructures()
 
     Listeners::g_dispatcher = this;
 
-    buildxlPolicyOps_ =
+    buildXLPolicyOps_ =
```
Comment (erikma) [inline Public/src/Sandbox/MacOs/BuildXLSandbox/Src/BuildXLSandbox.cpp]: Need to port renames like this back into the internal repo, rename branch

File: Public/src/Sandbox/MacOs/BuildXLSandbox/darwin/security/mac_policy.h [hunk 2]
```diff
@@ -6163,13 +6163,13 @@ typedef void mpo_vnode_notify_truncate_t(
 
 
 /**
-  @brief Inform MAC policies that a pty worker has been granted
+  @brief Inform MAC policies that a pty slave has been granted
```
Comment (erikma) [inline Public/src/Sandbox/MacOs/BuildXLSandbox/darwin/security/mac_policy.h]: Revert here and below. PoliCheck item

File: Public/src/Sandbox/MacOs/BuildXLSandbox/darwin/security/mac_policy.h [hunk 3]
```diff
@@ -3084,7 +3084,7 @@ typedef int mpo_proc_check_signal_t(
   @warning Caution should be exercised when implementing access
   controls for wait, since programs often wait for child processes to
   exit.  Failure to be notified of a child process terminating may
-  cause the parent process to stop responding, or may produce zombie processes.
+  cause the parent process to hang, or may produce zombie processes.
```
Comment (erikma) [inline Public/src/Sandbox/MacOs/BuildXLSandbox/darwin/security/mac_policy.h]: Revert this change

File: Public/src/Sandbox/MacOs/tools/iosimulator/Extensions.cs [hunk 4]
```diff
@@ -1,39 +1,36 @@
...
-                }
-            }
-            finally
-            {
-                Array.ForEach(enumerators, e => e.Dispose());
-            }
-        }
-    }
-}
+using System;
```
Comment (erikma) [inline Public/src/Sandbox/MacOs/tools/iosimulator/Extensions.cs]: Add open source comment header

File: Public/src/Sandbox/MacOs/tools/iosimulator/Hashing.cs [hunk 5]
```diff
@@ -1,6 +1,3 @@
-// Copyright (c) Microsoft. All rights reserved.
-// Licensed under the MIT license. See LICENSE file in the project root for full license information.
-
 using System;
```
Comment (erikma) [inline Public/src/Sandbox/MacOs/tools/iosimulator/Hashing.cs]: Restore open source header

File: Public/src/Sandbox/MacOs/tools/iosimulator/IOSimulatorApp.cs [hunk 6]
```diff
@@ -1,7 +1,4 @@
-// Copyright (c) Microsoft. All rights reserved.
-// Licensed under the MIT license. See LICENSE file in the project root for full license information.
-
-using System;
+﻿using System;
```
Comment (erikma) [inline Public/src/Sandbox/MacOs/tools/iosimulator/IOSimulatorApp.cs]: Restore open source header

File: Public/src/Sandbox/MacOs/tools/iosimulator/OutputDirectortWatcher.cs [hunk 7]
```diff
@@ -1,6 +1,3 @@
-// Copyright (c) Microsoft. All rights reserved.
-// Licensed under the MIT license. See LICENSE file in the project root for full license information.
-
 using System;
```
Comment (erikma) [inline Public/src/Sandbox/MacOs/tools/iosimulator/OutputDirectortWatcher.cs]: restore header

File: Public/src/Sandbox/MacOs/tools/iosimulator/Extensions.cs (hunk 4 above)
Comment (erikma) [reply Public/src/Sandbox/MacOs/tools/iosimulator/Extensions.cs]: Still need header here

File: Public/src/Demos/ReportAccesses/FileAccessReporter.cs [hunk 8]
```diff
@@ -48,6 +48,7 @@ public Task<SandboxedProcessResult> RunProcessUnderSandbox(string pathToProcess,
                         WorkingDirectory = workingDirectory,
                         PipSemiStableHash = 0,
                         PipDescription = "Simple sandbox demo",
+                        SandboxedKextConnection = OperatingSystemHelper.IsUnixOS ? new SandboxedKextConnection(numberOfKextConnections: 2) : null
```
Comment (smera) [inline Public/src/Demos/ReportAccesses/FileAccessReporter.cs]: >SandboxedKextConnection [](start = 24, length = 23)

Any relevant comments you think should be added to the demo docs?

File: Public/src/Sandbox/MacOs/.vscode/c_cpp_properties.json [hunk 9]
```diff
@@ -0,0 +1,80 @@
+{
+    "configurations": [
```
Comment (smera) [inline Public/src/Sandbox/MacOs/.vscode/c_cpp_properties.json]: >onfigur [](start = 6, length = 7)

is this a VSCode specific file? Do we need this?

File: Public/src/Sandbox/Windows/DetoursServices/FileAccessHelpers.h [hunk 10]
```diff
@@ -151,7 +151,12 @@ inline RequestedAccess operator|(RequestedAccess a, RequestedAccess b)
 // Represents the result of performing an access check (applying a PolicyResult to a proposed access and context such as file existence).
 class AccessCheckResult 
 {
+private:
+	AccessCheckResult() {}
+
 public:
+	static inline AccessCheckResult Invalid() { return AccessCheckResult(); }
+
```
Comment (smera) [inline Public/src/Sandbox/Windows/DetoursServices/FileAccessHelpers.h]: Just curious. Why did we have to add these? They were not part of the original code?

File: Public/src/Demos/ReportAccesses/FileAccessReporter.cs (hunk 8 above)
Comment (blufugen) [reply Public/src/Demos/ReportAccesses/FileAccessReporter.cs]: Yeah, this was on my list, thanks for reminding me.

File: Public/src/Sandbox/Windows/DetoursServices/FileAccessHelpers.h (hunk 10 above)
Comment (blufugen) [reply Public/src/Sandbox/Windows/DetoursServices/FileAccessHelpers.h]: This is left over from some source overwriting. Will remove.
